### ISSUE TITLE
[false-negative]ICAST_INTEGER_MULTIPLY_CAST_TO_LONG can not detect only contain left shift that may overflow

### ISSUE THREAD

**Issue by wuchiuwong** [spotbugs/spotbugs#1146]
**Affects SpotBugs Version:** 
spotbugs-4.0.0-RC2

**Rule:**
ICAST_INTEGER_MULTIPLY_CAST_TO_LONG

**Description:**
Unable to detect only contain left shift that may overflow, for example: <code>long foo = 1 << bar;</code>
This rule is implemented in SpotBugs.jar\main\java\edu\umd\cs\findbugs\detect\FindPuzzlers.java,
This rule only detects the case of multiplication overflow, and shift may also cause overflow, so it is recommended to supplement the detection of shift.

**Code Sample demonstrating the issue:**
```
denominator = 1 << this.puzzleDenominator;

private static long scanPart (String str, int start)
{
    int i = start;
    long val = -1;
    int len = str.length();
    while (i < len)
      {
        char ch = str.charAt(i);
        i++;
        int dig = Character.digit(ch, 10);
        if (dig < 0)
          {
            if (val < 0) return start << 16;
            return (val << 32) | (i << 16) | ((int) ch);
          }
        val = val < 0 ? dig : 10 * val + dig;
        if (val > Integer.MAX_VALUE)
          return -1; // overflow
      }
    return val < 0 ? (start << 16) : -1; // miss detect here
}
```

**Expected outcome:**
false-negative

**Running SpotBugs through:**
command line
